internal/controller/pkg/revision/deployment.go

```diff
@@ -110,6 +112,35 @@ func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRe
 			},
 		},
 	}
+	if revision.GetWebhookTLSSecretName() != nil {
+		v := corev1.Volume{
+			Name: "webhook-tls-secret",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: *revision.GetWebhookTLSSecretName(),
+					Items: []corev1.KeyToPath{
+						{Key: "tls.crt", Path: "tls.crt"},
+						{Key: "tls.key", Path: "tls.key"},
+					},
+				},
+			},
+		}
+		d.Spec.Template.Spec.Volumes = append(d.Spec.Template.Spec.Volumes, v)
+
+		vm := corev1.VolumeMount{
+			Name:      "webhook-tls-secret",
+			ReadOnly:  true,
+			MountPath: "/webhook/tls",
```

Comment:
nitpick: these strings (and a number of other values) that are used multiple times could be `const`.